pkg/controllers/monitoring-stack/controller.go

```diff
@@ -164,7 +166,41 @@ func (r *reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 
 		}
 	}
+	return r.cleanupResources(ctx, ms)
+}
+
+func (r *reconciler) cleanupResources(ctx context.Context, ms *stack.MonitoringStack) (ctrl.Result, error) {
+	if ms.ObjectMeta.DeletionTimestamp.IsZero() {
+		if !containsString(ms.ObjectMeta.Finalizers, finalizerName) {
+			controllerutil.AddFinalizer(ms, finalizerName)
+			if err := r.k8sClient.Update(ctx, ms); err != nil {
+				if errors.IsConflict(err) {
+					return ctrl.Result{RequeueAfter: 2 * time.Second}, nil
+				}
+				return ctrl.Result{}, err
+			}
+		}
+	} else if containsString(ms.ObjectMeta.Finalizers, finalizerName) {
+		grafanaDS := &grafanav1alpha1.GrafanaDataSource{}
+		if err := r.k8sClient.Get(ctx,
+			types.NamespacedName{Namespace: grafana_operator.Namespace, Name: fmt.Sprintf("ms-%s-%s", ms.Namespace, ms.Name)},
```

Comment:
I think we should have a `grafanaDSNameForStack(name, ns string) string` that returns the grafana-datasource name given a monitoring-stack's name and namespace so that this can be reused everywhere

Reply:
Done